src/site/_filters/paginate.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Take array of paginated hrefs and select the pages to display.
+ * @param {string[]} pages An array of hrefs of all the pages.
+ * @param {string} current Href of current page.
+ * @return {object[]} An array of up to 10 items to display, including href and index.
+ */
+module.exports = function paginate(pages, current) {
+  const currentIndex = pages.findIndex((page) => page === current);
+
+  if (currentIndex < 0) return [];
+
+  let curated = pages.slice(0, 10);
+  const start = currentIndex > 5 ? currentIndex - 5 : 0;
+  const end = currentIndex + 5;
+
+  if (currentIndex > 5) curated = pages.slice(start, end);
```

Comment:
It might be more readable, if you have this condition checked only once, eg:

let start = 0;
let end = 10;
if (currentIndex > 5) {
  start = currentIndex - 5;
  end = currentIndex + 5;
}
const curated = pages.slice(start, end);

You can also consider a more descriptive name than "curated", e,g, pagesToShow.